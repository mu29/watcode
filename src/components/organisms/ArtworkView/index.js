import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'
import { Heading, Tag, Text, IconButton } from 'components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
  width: 44rem;
  margin: 4rem 0;
  background-color: ${palette('white.default')};
`

const ImageArea = styled.div`
  position: relative;
  width: 10.5rem;
  height: 15rem;
  overflow: hidden;
  border: 0.0625rem solid ${palette('gray.40')};
`

const Image = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  filter: blur(0.1875rem);
`

const CodeArea = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 0.5rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.9);
`

const InfoArea = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  padding: 0 1rem;
`

const TagArea = styled.div``

const Title = styled(Heading)`
  margin-bottom: 0.25rem;
`

const StyledTag = styled(Tag)`
  margin: 0.1875rem 0.375rem 0.1875rem 0;
`

const TypeTag = styled(StyledTag)`
  font-weight: 400;
  color: ${palette('white.default')};
  background-color: ${palette('gray.90')};
`

const StyledText = styled(Text)`
  margin-left: 0.25rem;
`

const StyledIconButton = styled(IconButton)`
  margin: 1rem 0;
  color: ${palette('gray.90')};
  border-radius: 0;
  border-top: 0.0625rem solid ${palette('gray.30')};
  border-bottom: 0.0625rem solid ${palette('gray.30')};
  background-color: transparent;
`

const ArtworkView = ({
  artwork: {
    code,
    artist,
    title,
    type,
    tags,
    imageUrl,
  },
  ...props
}) => (
  <Wrapper { ...props }>
    <ImageArea>
      <Image src={ imageUrl } />
      <CodeArea>
        <Heading level={ 3 }>#{ code }</Heading>
        <Text color="gray.60" fontSize={ 12 } fontWeight={ 200 } small>by { artist }</Text>
      </CodeArea>
    </ImageArea>
    <InfoArea>
      <Title level={ 3 }>{ title }</Title>
      <TagArea>
        <TypeTag>{ type }</TypeTag>
        { tags.map(tag => (<StyledTag key={ tag }>{ tag }</StyledTag>)) }
      </TagArea>
      <StyledIconButton icon="star">
        <StyledText color={ null }>1293</StyledText>
      </StyledIconButton>
    </InfoArea>
  </Wrapper>
)

ArtworkView.propTypes = {
  artwork: PropTypes.shape({
    code: PropTypes.number,
    artist: PropTypes.string,
    title: PropTypes.string,
    tags: PropTypes.array,
    imageUrl: PropTypes.string,
  }).isRequired,
}

export default ArtworkView
