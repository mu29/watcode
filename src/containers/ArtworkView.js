import React from 'react'
import { connect } from 'react-redux'
import { ArtworkView } from 'components'

const ArtworkViewContainer = props => <ArtworkView { ...props } />

const mapStateToProps = () => ({
  artwork: {
    id: 112047,
    artist: 'nagatsuki taffee',
    title: 'Re: 제로부터 시작하는 이세계 생활',
    type: 'Light Novel',
    tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
    imageUrl: 'https://laftelimage.blob.core.windows.net/items/thumbs/large/3390a970-59de-4d04-a705-6814d2f2446a.jpg',
  },
})

export default connect(mapStateToProps)(ArtworkViewContainer)
