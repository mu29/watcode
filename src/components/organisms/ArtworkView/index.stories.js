import React from 'react'
import { storiesOf } from '@storybook/react'
import ArtworkView from '.'

const artwork = {
  code: 112047,
  artist: 'nagatsuki taffee',
  title: 'Re: 제로부터 시작하는 이세계 생활',
  type: 'Light Novel',
  tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
  imageUrl: 'https://laftelimage.blob.core.windows.net/items/thumbs/large/3390a970-59de-4d04-a705-6814d2f2446a.jpg',
}

storiesOf('Organisms/ArtworkView', module)
  .add('기본', () => (
    <ArtworkView artwork={ artwork } />
  ))
  .add('이미지 보이기', () => (
    <ArtworkView artwork={ artwork } showImage />
  ))
