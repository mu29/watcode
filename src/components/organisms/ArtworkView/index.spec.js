import React from 'react'
import ArtworkView from '.'

const artwork = {
  code: 112047,
  artist: 'nagatsuki taffee',
  title: 'Re: 제로부터 시작하는 이세계 생활',
  tags: ['rebirth', 'loop', 'harem', 'this is a long tag'],
  imageUrl: 'https://cdn.namuwikiusercontent.com/s/3f75b87a3b69db1af44479a12315ed58c7e6121c2842cf929209cbf74fa23b7b214a9124e7e494f6ffc8312bdcf231d53278bdf5e7330fb8b23c59ac2ebaf299a3775c6479795372d4b4be0533d58d63?e=1535506665&k=FZxNDPx_iAUdg9ZFqsqXGg',
}

const component = (props = {}) => withTheme(<ArtworkView artwork={ artwork } { ...props } />)

describe('<ArtworkView />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })
})
