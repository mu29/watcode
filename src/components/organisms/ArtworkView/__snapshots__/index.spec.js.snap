// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtworkView /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 44rem;
  margin: 4rem 0;
  background-color: #FFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  position: relative;
  width: 10.5rem;
  height: 15rem;
  overflow: hidden;
  border: 0.0625rem solid;
}

.c3 {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-filter: blur(0.1875rem);
  filter: blur(0.1875rem);
}

.c4 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.9);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 1rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        class="c3"
        src="https://cdn.namuwikiusercontent.com/s/3f75b87a3b69db1af44479a12315ed58c7e6121c2842cf929209cbf74fa23b7b214a9124e7e494f6ffc8312bdcf231d53278bdf5e7330fb8b23c59ac2ebaf299a3775c6479795372d4b4be0533d58d63?e=1535506665&k=FZxNDPx_iAUdg9ZFqsqXGg"
      />
      <div
        class="c4"
      >
        <span
          componentname="Heading"
        >
          #112047
        </span>
        <span
          componentname="Text"
        >
          by nagatsuki taffee
        </span>
      </div>
    </div>
    <div
      class="c5"
    >
      <span
        componentname="Heading"
      >
        Re: 제로부터 시작하는 이세계 생활
      </span>
      <div
        class=""
      >
        <span
          componentname="Tag"
        />
        <span
          componentname="Tag"
        >
          rebirth
        </span>
        <span
          componentname="Tag"
        >
          loop
        </span>
        <span
          componentname="Tag"
        >
          harem
        </span>
        <span
          componentname="Tag"
        >
          this is a long tag
        </span>
      </div>
      <span
        componentname="IconButton"
      >
        <span
          componentname="Text"
        >
          1293
        </span>
      </span>
    </div>
  </div>
  <span
    componentname="CommentList"
  />
</div>
`;
